deepchem/models/torch_models/layers.py

```diff
@@ -207,3 +205,266 @@ def forward(self,
+               ff_dropout_p: float, encoder_hsize: int,
+               encoder_dropout_p: float):
+    """Initialize a MATEncoder layer.
+
+    Parameters
+    ----------
+    dist_kernel: str
+      Kernel activation to be used. Can be either 'softmax' for softmax or 'exp' for exponential, for the self-attention layer.
+    lambda_attention: float
+      Constant to be multiplied with the attention matrix in the self-attention layer.
+    lambda_distance: float
+      Constant to be multiplied with the distance matrix in the self-attention layer.
+    h: int
+      Number of attention heads for the self-attention layer.
+    sa_hsize: int
+      Size of dense layer in the self-attention layer.
+    sa_dropout_p: float
+      Dropout probability for the self-attention layer.
+    output_bias: bool
+      If True, dense layers will use bias vectors in the self-attention layer.
+    d_input: int
+      Size of input layer in the feed-forward layer.
+    d_hidden: int
+      Size of hidden layer in the feed-forward layer.
+    d_output: int
+      Size of output layer in the feed-forward layer.
+    activation: str
+      Activation function to be used in the feed-forward layer.
+      Can choose between 'relu' for ReLU, 'leakyrelu' for LeakyReLU, 'prelu' for PReLU,
+      'tanh' for TanH, 'selu' for SELU, 'elu' for ELU and 'linear' for linear activation.
+    n_layers: int
+      Number of layers in the feed-forward layer.
+    dropout_p: float
+      Dropout probability in the feeed-forward layer.
+    encoder_hsize: int
+      Size of Dense layer for the encoder itself.
+    encoder_dropout_p: float
+      Dropout probability for connections in the encoder layer.
+    """
+    super(MATEncoderLayer, self).__init__()
+    self.self_attn = MultiHeadedMATAttention(dist_kernel, lambda_attention,
+                                             lambda_distance, h, sa_hsize,
+                                             sa_dropout_p, output_bias)
+    self.feed_forward = PositionwiseFeedForward(
+        d_input, d_hidden, d_output, activation, n_layers, ff_dropout_p)
+    layer = SublayerConnection(size=encoder_hsize, dropout_p=encoder_dropout_p)
+    self.sublayer = nn.ModuleList([layer for _ in range(2)])
+    self.size = encoder_hsize
+
+  def forward(self, x: torch.Tensor, mask: torch.Tensor, sa_dropout_p: float,
+              adj_matrix: np.ndarray, distance_matrix: np.ndarray):
+    """Output computation for the MATEncoder layer.
+
+    Parameters
+    ----------
+    x: torch.Tensor
+      Input tensor.
+    mask: torch.Tensor
+      Masks out padding values so that they are not taken into account when computing the attention score.
+    sa_dropout_p: float
+      Dropout probability for the self-attention layer (MultiHeadedMATAttention).
+    adj_matrix: np.ndarray
+      Adjacency matrix of a molecule.
+    distance_matrix: np.ndarray
+      Distance matrix of a molecule.
+    """
+    x = self.sublayer[0](x,
```

Comment:
Can you add a comment explaining what this does?

Reply:
I've added a small explanation to the docstring for the forward function.